components/uniset/src/uniset.rs

```diff
@@ -183,6 +188,93 @@ impl UnicodeSet {
             None => false,
         }
     }
+
+    /// Check if the calling Unicodeset contains all the characters of the given UnicodeSet
+    ///
+    /// # Example:
+    ///
+    /// ```
+    /// use icu_unicodeset::UnicodeSet;
+    /// let example_list = vec![65, 70, 85, 91]; // A - E, U - Z
+    /// let example = UnicodeSet::from_inversion_list(example_list).unwrap();
+    /// let a_to_d = UnicodeSet::from_inversion_list(vec![65, 69]).unwrap();
+    /// let f_to_t = UnicodeSet::from_inversion_list(vec![70, 85]).unwrap();
+    /// let r_to_x = UnicodeSet::from_inversion_list(vec![82, 88]).unwrap();
+    /// assert!(example.contains_set(&a_to_d)); // contains all
+    /// assert!(!example.contains_set(&f_to_t)); // contains none
+    /// assert!(!example.contains_set(&r_to_x)); // contains some
+    /// ```
+    pub fn contains_set(&self, set: &UnicodeSet) -> bool {
+        if set.as_inversion_list().len() > self.inv_list.len() {
+            return false;
+        }
+        let s = set.as_inversion_list().len() as f32;
+        let u = self.inv_list.len() as f32;
+        if s * u.log2() < u {
```

Comment:
Why are you doing this logical fork?  Is it an optimization?  I'm a little weary of converting these numbers to f32 and then doing a log on them.

Reply:
It's an optimization of a search between `O(S * log(U))` vs `O(U)` where `S` is the argument and `U` is the caller.

If `S` is significantly smaller than `U`, then we can perform our search in `O(S * log(U))` by calling our binary search, and else, we just go along every range in `U` and then check to make sure every range in `S` has been covered. 
(E.g. `S = 2, U = 16 ==> 2 log 16 = 8`)

If we are against the idea of calling `contains_range` and don't want to repeat the code from `contains_range`, we can just do the `O(U)` approach.

Reply:
Most real-life UnicodeSets contain only a few segments, since related characters are grouped together in the code point space.  The biggest sets are probably the [language exemplar characters](https://unicode-org.github.io/cldr-staging/charts/37/by_type/core_data.alphabetic_information.main.html), since they are less likely to be grouped together.  However, even those sets are probably no more than 10-20 ranges.

The other issue is that although your `O(S * log(U))` algorithm is faster on paper, it could potentially have less memory locality.  You have to bounce around the whole array for each lookup, which could mean a weakened ability for the processor to cache intermediate results.  Walking straight forward through a buffer is more efficient than random access due to memory locality.

I'm not sure how expensive f32::log2 is itself.  You can get the integer part of the logarithm by looking at the exponent in the floating point bits.  I assume computing the fraction part of the logarithm requires a bit of churning.  If you want to keep the fork here, consider calling the integer [leading_zeros](https://doc.rust-lang.org/std/primitive.u32.html#method.leading_zeros) function, which is probably faster and smaller code.  It's not as accurate, but it shouldn't matter which algorithm we use near the boundary.

tl;dr: In my opinion, the linear-time search is simple, understandable, and very fast for all real-life uses.  But, I don't feel strongly and am happy to go with the two-algorithm solution if you think it's better.

Reply:
Agreed and changed regarding removal unless proven necessary/optimal.

I never considered the expense of calling `f32::log2`, and was more of an optimization for "fun" I guess :)